docs/developers/guard-accounts.md

```diff
@@ -0,0 +1,134 @@
+
+## Ledger
+
+There is no solution released for the **Ledger**. As soon as there is one, it will be documented and referenced here.
+
+# For developers
+
+If you are a developer and you understand what a transaction is, what fields it contains and how a signing of a transaction happens you may be interested by the technical aspects of the implementation.  can be achieved by adding another field in the transaction structure for a second signature, and defining the option to flag the transactions that are supposed to be executed/verified as multisig transactions. The option field in the transaction is currently used to differentiate between transactions where the signature should be verified over the marshaled transaction and transactions where the signature should be verified over the hash of the transaction, but additional options could be defined.
+
+The full activation of the extra protection requires two actions:
+
+**1. Register a Guardian**
+
+**2. Guard the account**
+
+After guarding the account, _almost_ every action the user takes should be co-signed by the guardian.
+
+:::note
+**Why not mandate that every transaction is co-signed by a guardian when an account is guarded?**
+
+This is done in order to not have the user locked out of his account (e.g user guards his account but the guardian also loses access to his account). For this reason, the user should not only be able to **_set up a new guardian_**, but also **_change the guardian_** if he wishes so (e.g when the guardian is compromised).
+:::
+
+## Registering & Changing the guardian
+
+
+
+### Register a Guardian
+
+In order to register a guardian a user has to set a **guardian** address by sending a ```SetGuardian``` transaction (described in built-in functions [here](/developers/built-in-functions#setguardian)). The guardian address becomes active after **20 epochs**, a period longer than the unbond time (**10 epochs**). The guardian address should be set into the account’s key-value store.
+
+### Change the Active Guardian
+
+Accounts will continue to be compromised. For unknown reasons, you may be confronted with the necessity to change the active guardian. The scope of account guarding is to protect users in case of a compromised account, but there are edge cases where it is uncertain whether a user or a scammer is changing the guardian, potentially leading to a scammer taking control of a user's funds. To address this issue, additional measures are necessary to allow users to protect their funds. One solution is the implement for an activation time for setting a new guardian. If the transaction is not confirmed by an existing guardian, the activation time would give the user a chance to take action before the new guardian is set. In case the guardian confirms the transaction, the change could be made immediately. This solution would ensure that users have control over their accounts and can protect their funds even in the face of unexpected changes.
```

Comment:
```suggestion
Accounts can still be compromised by phishing attacks or other different ways. For unknown reasons, you may be confronted with the necessity to change the active guardian. The scope of account guarding is to protect users in case of a compromised account, but there are edge cases where it is uncertain whether a user or a scammer is changing the guardian, potentially leading to a scammer taking control of a user's funds. To address this issue, additional measures are necessary to allow users to protect their funds. One solution is the implement an activation time for setting a new guardian. If the transaction is not confirmed by an existing guardian, the activation time would give the user a chance to take action before the new guardian is set. In case the guardian confirms the transaction, the change could be made immediately. This solution would ensure that users have control over their accounts and can protect their funds even in the face of unexpected changes.
```